google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketContentEditorPanel.java

```diff
@@ -155,10 +167,19 @@ private void showBlobTable() {
    * BlobListOption#prefix(String)} options. The prefix acts as the current directory for the blobs
    * we wish to fetch.
    */
-  // TODO(eshaul) should be done asynchronously and show loader in the UI
-  private List<Blob> getBlobsStartingWith(String prefix) {
-    return Lists.newArrayList(
-        bucket.list(BlobListOption.currentDirectory(), BlobListOption.prefix(prefix)).iterateAll());
+  private void loadBlobsStartingWith(String prefix, Consumer<List<Blob>> afterLoad) {
+    showLoader();
+    ThreadUtil.getInstance()
+        .executeInBackground(
+            () -> {
+              List<Blob> blobs =
+                  Lists.newArrayList(
+                      bucket
+                          .list(BlobListOption.currentDirectory(), BlobListOption.prefix(prefix))
+                          .iterateAll());
+              hideLoader();
```

Comment:
Shouldn't this be done on the EDT as well? Not sure how thread-safe JPanel methods are.

Reply:
fixed